README.md

```diff
@@ -1 +1,114 @@
-# beta-subject-builder
\ No newline at end of file
+SubjectBuilderBlobTrigger - Python
+=========================================
+The subject builder creates a subject lookup table in a cosmos db instance.
+
+### TODO PostcodeSearchBuilder function processing steps
+
+1. Subject builder is triggered when a new blob (subjects) are stored in a storage container. It takes the path to the file and extracts the csv file.
+
+2. The function validates the header row of csv file to check it has the expected column headers before recreating the subjects collection in cosmos db.
+
+3. Following the recreation of the collection, the function iterates all subjects and stores them in the datastore.
+
+This process takes less than a minute to complete.
+
+### Subject data source
+
+The subject data that is uploaded to an azure storage container to which the azure function is configured to be triggered on is an xslx file that is with the OfS team.
+
+The xlsx file should contain the following header row (this is validated in the azure function):
+
+| code        | english_name | level | welsh_name |
+| ----------- | ------------ | ----- | -----------|
+| CAH20-01-01 | History      | 3     | Hanes      |
+| {n}         | ...          | ...   | ...        |
+
+There are 226 subject codes - these will need to be updated manually whereby the OfS team give the file to someone with access to the azure subscription to load the file into the azure storage container that the `SubjectBuilderBlobTrigger` function triggers on.
+
+### TODO Configuration Settings
+
+Add the following to your local.settings.json:
+
+| Variable                            | Default                | Description                                                  |
+| ----------------------------------- | ---------------------- | ------------------------------------------------------------ |
+| FUNCTIONS_WORKER_RUNTIME            | python                 | The programming language the function worker runs on         |
+| AzureWebJobsStorage                 | {retrieve from portal} | The default endpoint to access storage account               |
+| StopEtlPipelineOnWarning            | false                  | Boolean flag to stop function worker on a warning            |

```

Comment:
This exception is declared but not used. I would remove it from code and README.